Sources/SwiftParser/StringLiterals.swift

```diff
@@ -595,10 +595,105 @@ extension Parser {
       )
     }
   }
+
+  mutating func parseSimpleString() -> RawSimpleStringLiteralExprSyntax {
+    let openDelimiter = self.consume(if: .rawStringPoundDelimiter)
+    let (unexpectedBeforeOpenQuote, openQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.OpeningQuoteOptions.self, default: .stringQuote)
+
+    /// Parse segments.
+    var segments: [RawStringSegmentSyntax] = []
+    var loopProgress = LoopProgressCondition()
+    while hasProgressed(&loopProgress) {
+      // If we encounter a token with leading trivia, we're no longer in the
+      // string literal.
+      guard currentToken.leadingTriviaText.isEmpty else { break }
+
+      if let stringSegment = self.consume(if: .stringSegment, TokenSpec(.identifier, remapping: .stringSegment)) {
+        var unexpectedAfterContent: RawUnexpectedNodesSyntax?
+
+        if let (backslash, leftParen) = self.consume(if: .backslash, followedBy: .leftParen) {
+          var unexpectedTokens: [RawSyntax] = [RawSyntax(backslash), RawSyntax(leftParen)]
+
+          let (unexpectedBeforeRightParen, rightParen) = self.expect(TokenSpec(.rightParen, allowAtStartOfLine: false))
+          unexpectedTokens += unexpectedBeforeRightParen?.elements ?? []
+          unexpectedTokens.append(RawSyntax(rightParen))
+
+          unexpectedAfterContent = RawUnexpectedNodesSyntax(
+            unexpectedTokens,
+            arena: self.arena
+          )
+        }
+
+        segments.append(RawStringSegmentSyntax(content: stringSegment, unexpectedAfterContent, arena: self.arena))
+      } else {
+        break
+      }
+    }
+
+    let (unexpectedBetweenSegmentAndCloseQuote, closeQuote) = self.expect(
+      anyIn: SimpleStringLiteralExprSyntax.ClosingQuoteOptions.self,
+      default: openQuote.closeTokenKind
+    )
+    let closeDelimiter = self.consume(if: .rawStringPoundDelimiter)
+
+    if openQuote.tokenKind == .multilineStringQuote, !openQuote.isMissing, !closeQuote.isMissing {
+      let postProcessed = postProcessMultilineStringLiteral(
+        rawStringDelimitersToken: openDelimiter,
+        openQuote: openQuote,
+        segments: segments.compactMap { RawStringLiteralSegmentListSyntax.Element.stringSegment($0) },
+        closeQuote: closeQuote
+      )
+
+      return RawSimpleStringLiteralExprSyntax(
+        RawUnexpectedNodesSyntax(
+          combining: unexpectedBeforeOpenQuote,
+          postProcessed.unexpectedBeforeOpeningQuote,
+          openDelimiter,
+          arena: self.arena
+        ),
+        openingQuote: postProcessed.openingQuote,
+        segments: RawSimpleStringLiteralSegmentListSyntax(
+          elements: postProcessed.segments.compactMap { $0.as(RawStringSegmentSyntax.self) },
```

Comment:
The reason why we can do the `compactMap` here is that we know we added only `RawStringSegmentSyntax` to the segments above, right? I think we should call this out in a short comment. At first it reads like we might be dropping syntax nodes here. And if we can guarantee that `postProcessed.segments` only contains `RawStringSegmentSyntax`, this can be the following, which makes the intent clearer

```suggestion
          elements: postProcessed.segments.map { $0.cast(RawStringSegmentSyntax.self) },
```

Reply:
We can guarantee that. It will not change the type of the segments.

Reply:
The method `.cast` is not available. 
`RawSyntaxNodeProtocol` only have `as` and `is`